src/objectExplorer/objectExplorerService.ts

```diff
@@ -396,7 +396,7 @@ export class ObjectExplorerService {
             nodePath: node.nodePath
         };
         await this._connectionManager.client.sendRequest(RefreshRequest.type, refreshParams);
-        return this._objectExplorerProvider.refresh(node);
+        return this._objectExplorerProvider.refresh(undefined);
```

Comment:
Why wouldn't you just refresh the tree after the node is deleted? This seems really heavy handed to refresh the whole tree every time we call refreshNode (which given the name would imply to me that it would only refresh the node itself)

Reply:
That's what I'm doing, refreshing the tree in the end. The `refreshNode` here means the refresh from backend. The `this._objectExplorerProvider.refresh(undefined)` is refreshing the Tree View. They are unrelated.

Reply:
What I'm saying is that this function appears to be called when you right click on a node and select refresh. In that case I don't think we should be refreshing the whole tree - that's extremely inefficient.

Why wouldn't you just refresh the tree after deleting a node? Like have the code that's actually deleting the node call refresh on the tree afterwards.

Reply:
It already does. But the delete `XXX` node from UI is different from running `drop database XXX` from query editor.